setup.sh

```diff
@@ -126,25 +126,32 @@ echo "@@@@ Correction include: $CORRECTION_INCLUDE_DIR"
 echo "@@@@ Correction lib: $CORRECTION_LIB_DIR"
 
 # JSONPOG integration auto-update
-export JSONPOG_INTEGRATION_PATH=$SKNANO_HOME/external/jsonpog-integration
-echo "@@@@ Checking for updates in jsonpog-integration repository..."
-cd "$JSONPOG_INTEGRATION_PATH"
-git fetch origin 
-LOCAL_HASH=$(git rev-parse HEAD) # local hash of latest
-REMOTE_HASH=$(git rev-parse origin/master) # remote hash of latest
-echo "@@@@ Local latest commit: $LOCAL_HASH"
-echo "@@@@ Remote latest commit: $REMOTE_HASH"
-
-# latest commit date and message of the remote branch
-REMOTE_DATE=$(git log -1 --format=%ci origin/master) 
-REMOTE_MESSAGE=$(git log -1 --format=%s origin/master) 
-echo "@@@@ Remote latest update: $REMOTE_DATE - $REMOTE_MESSAGE"
-cd -    
-if [ "$LOCAL_HASH" != "$REMOTE_HASH" ]; then
-    echo "@@@@ Updating jsonpog-integration repository..."
-    git submodule update --remote external/jsonpog-integration
+JSONPOG_REPO_PATH="$SKNANO_HOME/external/jsonpog-integration"
+
+if [ ! -d "$JSONPOG_REPO_PATH" ]; then
+    echo "@@@@ JSONPOG Repository not found"
 else
-    echo "@@@@ jsonpog-integration repository is already up-to-date."
+    echo "@@@@ Checking for updates in jsonpog-integration repository..."
+    cd "$JSONPOG_REPO_PATH"
+    git fetch origin
+
+    CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+
+    if [ -z "$(git symbolic-ref -q HEAD)" ]; then
+        echo "@@@@ HEAD is detached. Switching back to the previous branch: $CURRENT_BRANCH"
+        git checkout "$CURRENT_BRANCH"
+    fi
+
+    BEHIND=$(git rev-list --count origin/master..HEAD)
+
+    if [ "$BEHIND" -gt 0 ]; then
+        echo "@@@@ Repository is $BEHIND commits behind origin/master, updating..."
+        git merge origin/master
```

Comment:
Can you add logic for accept / reject the update? It is hard to track the changes if the jsonpog repo is auto-updating for every setup, especially for reproducing results or debugging the issue.